### DISCUSSION TITLE
Questions about doublecmd.xml file

### DISCUSSION THREAD

**Discussion by advice2020** [doublecmd/doublecmd#1907] [General]
1.) I noticed that you cannot make certain modifications to the doublecmd.xml file.
-I tried adding blank lines to visually separate some items.
You can save this but when you open DC and then close it the save removes them
-I also tried adding some "comments" `<!--comment--> `and when you do this and save, the next time you open up DC you get an error dialog box, so this is not allowed either.
I was wondering why this would be?

2.) I have not tried this yet because I do not know how to :) but I had an idea for something in the past and thought that maybe xml "includes" would be a possible solution for it.  I imagine if the things I mentioned above are not supported then "includes" would not be either?
Does anyone know if it is possibel to use "includes" in the doublecmd.xml file?
and if so how to do so?
I have seen a couple of ways mentioned on how to add includes into xml files but is there a specific way to do this in doublecmd.xml file?

Thank You

**Comment by Skif-off** [comment]
1.) DC reads the entire file and takes the necessary parameters from it. When saving, the file is generated anew with the parameters from memory and replaces the previous one.
This won't work.

2.) [For example](https://doublecmd.h1n.ru/viewtopic.php?t=8237).

**Comment by advice2020** [comment]
@Skif-off 
Thank you for your response
1.) That is unfortunate
2.) I was not familiar with "RedirectSection" yeah that appears simlar to what an "include" would offer.
3.) I added a third item in my original post above if interested kind of more of an observation/question
EDIT: just moved it below instead

**Comment by JonnyTech** [comment]
> 3.) I added a third item in my original post above if interested kind of more of an observation/question

@advice2020 please do not edit previous posts. We do not recieve notifications. And it makes the conversation difficult to follow. Kindly create a new post.

**Comment by advice2020** [comment]
@JonnyTech
(I made the edit before seeing your response)
Oh I apologize I didn't think any other users really even read any of my posts :)
Okay, so what should I do because it is about the same topic of doublecmd.xml file, thought it would be better to just keep all my posts in the same post.  I run into the same thing with request posts, if I make each one a separate post I would take over this entire forum :)  so I usually try to combine even though I agree more confusing.

**Comment by advice2020** [comment]
3.) Something to keep in mind.  I started making copies of my DC settings files (for example doublecmd.xml) a while back so that I can just overwrite them as newer versions come out.  Something that I did not think of is that as DC gets some updates, this causes the doublecmd.xml file change, containing these newer updates as well.  So if you overwrite with your older settings file you do not have access to those setting in the xml file if needed.
I learned this most recently with the addition of the amazing update of being able to D&D move by default, which is achieved by a setting that needs to be manually set in this xml file.
Even though I am aware that this occurs it does not stop me from doing it every now and then (which I will mention in a minute)
The only solution I could think of was to use the DC built in compare tool (which is amazing!).
In order to keep all your custom settings from one xml file to another, you have to take the new items from one and bring it over into your xml file and then save all the work you did and then you eventually close out of DC......
To then realize that in order to use the DC compare tool you need to open DC, but.... you are editing the doublecmd.xml file, which..... when you close DC, it.... saves over the work you did to that file :)
If anyone does this, some advice, it is a better idea to keep the new doublecmd.xml file in the settings folder and update your customized one either with a different name or in a different folder and then use that one to replace the doublecmd.xml file.
Anyways is this the best method of achieving this? to manually compare these xml files when you update to newer versions of DC?

Thank You

**Comment by JonnyTech** [comment]
If you really need DC to edit the config file: temporarily install a portable version of DC - it contains its own settings folder - and use that to edit the main DC's config file.

**Comment by JonnyTech** [comment]
> Okay, so what should I do because it is about the same topic of doublecmd.xml file, thought it would be better to just keep all my posts in the same post.

Guess we need to sort out terminology:

Issue = topic = thread

Post = comment = reply

So in GitHub terms: add a new comment to the issue (which is what you have done, thank you) so that subscribers will recieve a notification and you are more likely to get a reply.

**Comment by Skif-off** [comment]
> temporarily install a portable version of DC

[--config-dir=](https://doublecmd.github.io/doc/en/commandline.html)

**Comment by advice2020** [comment]
Thank you for the responses.
As mentioned I didn't think anyone subscribed to any of my posts, to be honest do not know much about that feature.  I also think I have an idea on how I will update threads in the future if I need to.

Also I will show you that terminology is not so easy around here :)
"So in GitHub terms: add a new comment to the issue"
In GitHub it has an "Issues" section, which is another use of the word "issue", well this is the "Discussions" section so don't you mean
"So in GitHub terms: add a new comment to the Discussion" :)

Thank you both again for the comments that contributed to my Discussion.... :)

**Comment by advice2020** [comment]
Sorry to bring this up again but realized I am still a little confused.
-When you customize DC all gets saved in doublecmd.xml file.
-If you save/backup this file with the intent to be used again in newer versions of DC alpha standalone version.
-When new alpha version is setup it creates a new default doublecmd.xml file which if any new settings are added to DC, these new features (code) is added to this xml file.
-If I then replace it with my older custom version, my version does not have the new feature code in it.
DC does not automatically update my custom xml putting new code (new settings) into my it, correct?
So I would have to make these updates manually, bringing the new code to my custom xml file?

Only way I thought of to correct this was to use DC compare tool with the two files, but my custom xml file contains tons of text compared to the new default xml file, so is very difficult to manually verifiy if any new changes have been made to DC default xml file and would have to do this any time decide to test out newer DC version.

Just wondering how other users / proper way to handle this?
Thank You

**Comment by Skif-off** [comment]
@advice2020 

> -When new alpha version is setup it creates a new default doublecmd.xml file which if any new settings are added to DC, these new features (code) is added to this xml file.

If new options are added, the current `doublecmd.xml` will be read, and the new options will be used with the default values. After closing the program, the new options/values will be written to the file.

> -If I then replace it with my older custom version, my version does not have the new feature code in it.

Show an example. The vast majority of parameters are written to `doublecmd.xml` when the program is closed.

**Comment by advice2020** [comment]
@Skif-off 
Ahh thank you, I had wondered if this would happen.
I do not remember this occurring when the drag "move" setting was added recently, which I think was one of the reasons for me making this post to begin with.
Now that I know this should occur, I will check if I can see this in action :)  if I have any older doublecmd.xml files saved without this setting.
This is good to know because could not think of an efficient way on how to handle this because it defeated the purpose of being able to save the xml file and use between versions.  Great news :)
Thank You
